temporal/fx.go

```diff
@@ -612,144 +606,137 @@ func ApplyClusterMetadataConfigProvider(
-		if clusterName != clusterData.CurrentClusterName {
-			logger.Warn(
-				"ClusterInformation in ClusterMetadata config is deprecated. "+
-					"Please use TCTL admin tool to configure remote cluster connections",
-				tag.Key("clusterInformation"),
-				tag.ClusterName(clusterName),
-				tag.IgnoredValue(clusterInfo))
-			continue
-		}
+	// Only configure current cluster metadata from static config file
+	clusterId := uuid.New()
+	applied, err := clusterMetadataManager.SaveClusterMetadata(
+		ctx,
+		&persistence.SaveClusterMetadataRequest{
+			ClusterMetadata: persistencespb.ClusterMetadata{
+				HistoryShardCount:        config.Persistence.NumHistoryShards,
+				ClusterName:              clusterData.CurrentClusterName,
+				ClusterId:                clusterId,
+				ClusterAddress:           clusterData.InternalRPCAddress,
+				FailoverVersionIncrement: clusterData.FailoverVersionIncrement,
+				InitialFailoverVersion:   clusterData.InitialFailoverVersion,
+				IsGlobalNamespaceEnabled: clusterData.EnableGlobalNamespace,
+				IsConnectionEnabled:      true, // current cluster connection always be enabled
+			},
+		})
+	if err != nil {
+		logger.Warn("Failed to save cluster metadata.", tag.Error(err), tag.ClusterName(clusterData.CurrentClusterName))
+	}
+	if applied {
+		logger.Info("Successfully saved cluster metadata.", tag.ClusterName(clusterData.CurrentClusterName))
+		return config.ClusterMetadata, config.Persistence, nil
+	}
+	resp, err := clusterMetadataManager.GetClusterMetadata(ctx, &persistence.GetClusterMetadataRequest{
+		ClusterName: clusterData.CurrentClusterName,
+	})
+	if err != nil {
+		return config.ClusterMetadata, config.Persistence, fmt.Errorf("error while fetching cluster metadata: %w", err)
+	}
+	err = updateClusterData(ctx, clusterMetadataManager, resp, clusterData)
+	if err != nil {
+		return config.ClusterMetadata, config.Persistence, err
+	}
+	overwriteClusterData(resp, clusterData, config, logger)
+	return config.ClusterMetadata, config.Persistence, nil
+}
 
-		// Only configure current cluster metadata from static config file
-		clusterId := uuid.New()
+func updateClusterData(
+	ctx context.Context,
+	clusterMetadataManager persistence.ClusterMetadataManager,
+	persistedData *persistence.GetClusterMetadataResponse,
+	clusterData *cluster.Config,
+) error {
+	// Allow updating current cluster metadata if global namespace is disabled
+	updateCurrentClusterMetadata := false
+	currentMetadata := persistedData.ClusterMetadata
+	if !persistedData.IsGlobalNamespaceEnabled && clusterData.EnableGlobalNamespace {
+		currentMetadata.IsGlobalNamespaceEnabled = clusterData.EnableGlobalNamespace
+		currentMetadata.InitialFailoverVersion = clusterData.InitialFailoverVersion
+		currentMetadata.FailoverVersionIncrement = clusterData.FailoverVersionIncrement
+		updateCurrentClusterMetadata = true
+	}
+	// Allow updating current cluster internal rpc address
+	if persistedData.ClusterAddress != clusterData.InternalRPCAddress {
```

Comment:
If InternalRPCAddress is not set, will this remove existing ClusterAddress from DB?
I think we should only do this if len(clusterData.InternalRPCAddress) > 0